filebeat/harvester/log.go

```diff
@@ -47,9 +50,42 @@ func NewHarvester(
 	return h, nil
 }
 
+func createLineReader(
+	in FileSource,
+	codec encoding.Encoding,
+	bufferSize int,
+	maxBytes int,
+	readerConfig logFileReaderConfig,
+	mlrConfig *config.MultilineConfig,
+) (processor.LineProcessor, error) {
+	var p processor.LineProcessor
+	var err error
+
+	fileReader, err := newLogFileReader(in, readerConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	p, err = processor.NewLineSource(fileReader, codec, bufferSize)
+	if err != nil {
+		return nil, err
+	}
+
+	if mlrConfig != nil {
+		p, err = processor.NewMultiline(p, maxBytes, mlrConfig)
+		if err != nil {
+			return nil, err
+		}
+
+		return processor.NewStripNewline(p), nil
+	}
+
+	p = processor.NewStripNewline(p)
+	return processor.NewLimitProcessor(p, maxBytes), nil
```

Comment:
You add here Processor to the name but we don't do it for the other processors. Should we call it just NewLimit(...)?